app/javascript/components/App.js

```diff
@@ -19,6 +19,13 @@ import {
 
 
 class App extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      locations: [],
+      reviews: [],
+    }
+  }     
```

Comment:
I don't think you are going to need state for the header component.  This should be removed as well.